intermediate/challenge-1/src/main/java/com/audela/challenge/busapi/controller/BusController.java

```diff
@@ -0,0 +1,99 @@
+package com.audela.challenge.busapi.controller;
+
+
+import com.audela.challenge.busapi.entity.BusEntity;
+import com.audela.challenge.busapi.entity.DriverEntity;
+import com.audela.challenge.busapi.entity.ScheduleEntity;
+import com.audela.challenge.busapi.exception.DataValidationException;
+import com.audela.challenge.busapi.exception.ScheduleConflictException;
+import com.audela.challenge.busapi.service.BusService;
+import com.audela.challenge.busapi.vo.BusScheduleVo;
+import com.audela.challenge.busapi.vo.DriverScheduleVo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api")
+public class BusController {
+
+    @Autowired
+    private BusService busService;
+
+    @PostMapping(value = "/driver", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
From a code perspective  , how would you deal with a major API refactor and properly communicate the modification to the consumers to avoid service disruptions  ?

Reply:
We can keep the old api and decommission it in a later point.